src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs

```diff
@@ -131,10 +131,59 @@ public PackageArchiveReader BuildPackage(PackageBuilder builder, string outputPa
                 PrintVerbose(outputPath, builder);
             }
 
+            if (_packArgs.InstallPackageToOutputPath)
+            {
+                _packArgs.Logger.LogMinimal(String.Format(CultureInfo.CurrentCulture, Strings.Log_PackageCommandInstallPackageToOutputPath, "Package", outputPath));
+                WriteResolvedNuSpecToPackageOutputDirectory(outputPath, builder);
+                WriteSHA512PackageHash(outputPath);
+            }
+
             _packArgs.Logger.LogMinimal(String.Format(CultureInfo.CurrentCulture, Strings.Log_PackageCommandSuccess, outputPath));
             return new PackageArchiveReader(outputPath);
         }
 
+        /// <summary>
+        /// Writes the resolved NuSpec file to the package output directory.
+        /// </summary>
+        /// <param name="packageOutputPath">The output package path</param>
+        /// <param name="packageBuilder">The package builder</param>
+        private void WriteResolvedNuSpecToPackageOutputDirectory(string packageOutputPath, PackageBuilder packageBuilder)
+        {
+            string packageOutputDirectory = Path.GetDirectoryName(packageOutputPath);
+            string resolvedNuSpecOutputPath = Path.Combine(packageOutputDirectory, packageBuilder.Id + PackagingConstants.ManifestExtension);
```

Comment:
Use VersionFolderPath resolver to create these names

 https://github.com/NuGet/NuGet.Client/blob/2ea9d7ca853506797b891e736955fe8b47ff1793/src/NuGet.Core/NuGet.Packaging/VersionFolderPathResolver.cs#L171

Reply:
I had trouble with that, will spend more time with it and see if I can get the desired output.